app/models/illiad_request.rb

```diff
@@ -0,0 +1,89 @@
+###
+#  Class to handle creation of ILLiad OpenURL request
+###
+class IlliadRequest
+  def initialize(current_user, scan)
+    @current_user = current_user
+    @scan = scan
+  end
+
+  def illiad_transaction_response
+    # If the illiad_transaction_request has a missing required value (e.g. no Username) there will not be a response body
+    # ILLiad will simply return an empty response which we deal with in the ScanController check_illiad method
+    response.respond_to?(:body) ? response.body : response
```

Comment:
Ok, I see what you mean. Seems like trying to follow the pattern used for symphony was not exactly the way to go since illiad already returns an empty response. I simplified things and now the controller only deals with the same response body in all cases.